features/step_definitions/steps.js

```diff
@@ -109,3 +130,71 @@ When('you create your user settings via POST request to {string} with:', functio
   postRequest(route, JSON.stringify(jsonBody), callback);
 });
 
+Then('you will stop seeing posts of the user with id {string}', function (blacklistedUserId) {
+  return this.app.service('usersettings').find({userId: currentUser._id.toString()}).then((settings) => {
+    expect(settings.total).to.eq(1);
+    expect(settings.data[0].blacklist).to.be.an('array').that.does.include(blacklistedUserId);
+  });
+});
+
+Given('you blacklisted the user {string} before', async function (blacklistedUserName) {
+  const res = await this.app.service('users').find({query: {name: blacklistedUserName}});
+  blacklistedUser = res.data[0];
+  const params = {
+    userId: currentUser._id,
+    blacklist: [blacklistedUser._id]
+  }
+  return this.app.service('usersettings').create(params)
+});
+
+When('this user publishes a post', function () {
+  const params = {
+    title: 'Awful title',
+    content: 'disgusting content',
+    language: 'en',
+    type: 'post',
+    userId: blacklistedUser._id
+  }
+  return this.app.service('contributions').create(params)
+});
+
+When('you read your current news feed', function (callback) {
+  getRequest('/contributions', callback);
+});
+
+Then('this post is not included', function () {
+  expect(httpResponse.data).to.be.an('array').that.is.empty;
+});
+
+Given('there is a post {string} by user {string}', async function (postTitle, userName) {
+  const users = await this.app.service('users').find({ query: {name: userName} });
+  const user = users.data[0];
+  const params = {
+    title: postTitle,
+    content: 'blah',
+    language: 'en',
+    type: 'post',
+    userId: user._id
+  }
+  lastPost = await this.app.service('contributions').create(params)
+  return lastPost;
+});
+
+Given('the blacklisted user wrote a comment on that post:', function (comment) {
+  const commentParams = {
+    userId: blacklistedUser._id,
+    content: comment,
+    contributionId: lastPost._id
+  };
+  return this.app.service('comments').create(commentParams)
+});
+
+When('you read through the comments of that post', function (callback) {
+  getRequest('/comments', callback);
+});
+
+Then('you will see a hint instead of a comment:', function (hint) {
+  const comment = httpResponse.data[0];
+  expect(comment.content).to.eq(hint);
+  expect(comment.contentExcerpt).to.eq(hint);
```

Comment:
@appinteractive @sebako86 I created three cucumber scenarios:
1. Write blacklist into `usersettings`
2. News feed should not include blacklisted usercontent
3. Comments should conceal blacklisted comments

Regarding 3, do you want the API response to include the actual comment content and the frontend displays a computed `visibleContent` or do you want the server to respond with the displayed content only?

As you can see in the implementation of the test, I am assuming the latter solution.
The other solution would enable a feature to unhide concealed user comments on the client side.

Reply:
1. & 2. cool
3. no need to have it handled on client side, let's safe the bandwidth for now and fetch any concealed content on demand.